app/models/product.rb

```diff
@@ -1,6 +1,7 @@
 class Product < ApplicationRecord
-  has_many :line_items
+  has_many :line_items, dependent: :restrict_with_exception
```

Comment:
which is a better option here
`restrict_with_exception` or `restrict_with_error`